week-2-wide-perspective/w02e01-functional-warm-up/src/index.ts

```diff
@@ -0,0 +1,63 @@
+import {randomAlbumData} from "./data-store/music-album.js"
+import {fruitsData} from "./data-store/fruits";
+import {AlbumDataType, FruitsData, UserData} from "./type";
+import {usersData} from "./data-store/users";
+
+
+export const getAlbumData =<T extends AlbumDataType >(album: T )=> {
+    const genresName = album.genres[0].name
+    const artistName = album.artist.name
+    const tracksLength  = album.tracks.length
+    const totalAlbumDuration = album.tracks.reduce(((currentValue,track)=>currentValue + Number(track.duration) ),0)
+    const ShortestTrack = album.tracks.sort((a,b)=> Number(a.duration) - Number(b.duration))[0]
+
+ return `
+ genresName: ${genresName}
+  artistName: ${artistName}
+  tracksLength:(${tracksLength})
+   ShortestTrack - ${ShortestTrack.title}
+   totalAlbumDuration: ${totalAlbumDuration}`
+}
+
+
+export const getFruitData =<T extends FruitsData[]> (data: T) => {
+    const allFruitsFamily =  [...new Set(data.map(({family})=>  family ? family: null).filter(((item)=> item !== null)))]
+    const allFruitsStartsWithG = data.filter((({name})=> name.startsWith("G") && name)).map((({name})=> name))
+    const lowesCalories = data.sort(((A,B)=>  A.nutritions.calories - B.nutritions.calories)).map((({name,nutritions: {calories}})=> {
+        return {
+            name,
+            calories
+        }
+    }))[0]
+    return `
+     allFruitsFamily: ${allFruitsFamily}
+     allFruitsStartsWithG: ${allFruitsStartsWithG}
+     lowesCalories: fruitName: ${lowesCalories.name}, calories: ${lowesCalories.calories} kcal
+    `
```

Comment:
Generalnie @DragoBJJ zadanie jest zrobione dobrze. 

W senesie - o to chodziło. O wykorzystanie Fluent API dla `Array`.

Jednak - jak dla mnie to jest faza "przed refactoringiem" i szczegóły masz w ogólnym komentarzu.